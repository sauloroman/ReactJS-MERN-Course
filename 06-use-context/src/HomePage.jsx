export const HomePage = () => {
  return (
    <div>
      <h1 className="heading__page">Home Page</h1>
    </div>
  )
}
