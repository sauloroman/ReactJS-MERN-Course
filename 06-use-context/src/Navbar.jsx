import { NavLink } from 'react-router-dom'

export const Navbar = () => {
  return (
    <div className="nav">
      <ul className="nav__list">
        <NavLink to={'/'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>Home</li>
        </NavLink>
        <NavLink to={'/about'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>About</li>
        </NavLink>
        <NavLink to={'/login'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>Login</li>
        </NavLink>
      </ul>
    </div>
  )
}
