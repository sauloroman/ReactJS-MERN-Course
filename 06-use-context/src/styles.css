*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%;
}

body {
  box-sizing: border-box;
  font-size: 1.6rem;
  font-family: 'Poppins', sans-serif;
  color: #555;
}

.container {
  margin: 0 auto;
  width: 120rem;
  max-width: 95%;
}

.heading {
  font-size: 3.2rem;
  color: #5959ec;
  margin-bottom: 2rem;
}

.heading__page {
  font-size: 2rem;
  color: #000;
}

.nav {
  background-color: #000;
  margin-bottom: 2rem;
}

.nav__list {
  list-style: none;
  display: flex;
  justify-content: space-around;
}

.nav__item {
  width: 100%;
  text-align: center;
  padding: 1rem 0;
  color: #fff;
  text-decoration: none;
}

.btn {
  background-color: #5959ec;
  font-family: inherit;
  color: #fff;
  padding: 1rem 2rem;
  border: none;
  cursor: pointer;
}