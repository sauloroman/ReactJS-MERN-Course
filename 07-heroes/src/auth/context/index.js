export * from './AuthContext'
export * from './AuthProvider.jsx'
export * from './auth.reducer'