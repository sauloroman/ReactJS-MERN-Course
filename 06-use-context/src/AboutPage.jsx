import { useContext } from "react"
import { UserContext } from "./context/UserContext"

export const AboutPage = () => {

  const { user } = useContext(UserContext)

  return (
    <div>
      <h1 className="heading__page">About Page</h1>

      <div className="user">
        <p className="user__name">{user.name}</p>
        <p className="user__email">{user.email}</p>
        <p className="user__age">{user.age}</p>
      </div>
    </div>
  )
}
