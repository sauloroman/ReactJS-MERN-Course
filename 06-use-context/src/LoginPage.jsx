import { useContext } from "react"
import { UserContext } from "./context/UserContext"

export const LoginPage = () => {

  const { setUser } = useContext(UserContext)

  return (
    <div>
      <h1 className="heading__page">Bienvenido a la app</h1>
      <button 
        className="btn"
        onClick={() => setUser({ name: 'Saulo Román', email: 'romansantillan1998@outlook.com', age: 26, })}>Iniciar Sesión</button>
    </div>
  )
}
