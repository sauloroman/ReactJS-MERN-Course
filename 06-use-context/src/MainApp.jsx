import { Navigate, Route, Routes } from "react-router-dom"
import { Navbar } from "./Navbar"
import { HomePage } from "./HomePage"
import { AboutPage } from "./AboutPage"
import { LoginPage } from "./LoginPage"
import { UserProvider } from "./context/UserProvider"

export const MainApp = () => {
  return (
    <UserProvider>
      <div className="container">
        <h1 className="heading">UseContext</h1>
        <Navbar />

        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/about" element={<AboutPage />} />
          <Route path="/login" element={<LoginPage />} />
          <Route path="/*" element={<Navigate to={'/'}/>} />
        </Routes>
      </div>
    </UserProvider>
  )
}
