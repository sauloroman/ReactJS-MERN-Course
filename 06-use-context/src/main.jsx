import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { MainApp } from './MainApp'
import { BrowserRouter } from 'react-router-dom'
import './styles.css'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <BrowserRouter>
      <MainApp />
    </BrowserRouter>
  </StrictMode>,
)
