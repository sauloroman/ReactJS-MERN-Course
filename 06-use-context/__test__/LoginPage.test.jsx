import { render, screen, fireEvent } from '@testing-library/react'
import { UserContext } from '../src/context/UserContext'
import { LoginPage } from '../src/LoginPage'

describe('Tests in LoginPage Component', () => {

  test('it must match with snapshot', () => {

    const { container } = render(
      <UserContext.Provider value={{}}>
        <LoginPage />
      </UserContext.Provider>
    ) 

    expect( container ).toMatchSnapshot()

  })

  test('It must show the component at the initial state', () => {

    render(
      <UserContext.Provider value={{ user: null }}>
        <LoginPage />
      </UserContext.Provider>
    )

    const buttonToSetUser = screen.getByRole('button')
    expect( buttonToSetUser ).toBeTruthy()
    expect( buttonToSetUser.textContent ).toBe('Iniciar Sesión')

    const titleComponent = screen.getByRole('heading', { level: 1 })
    expect( titleComponent ).toBeTruthy()
    expect( titleComponent.textContent ).toBe('Bienvenido a la app')

  })

  test('It must call the function to set a user if button is pressed', () => {

    const setUserMock = jest.fn()

    render(
      <UserContext.Provider value={{ user: null, setUser: setUserMock }}>
        <LoginPage />
      </UserContext.Provider>
    )

    const button = screen.getByRole('button')
    fireEvent.click( button )

    expect( setUserMock ).toHaveBeenCalled()
    expect( setUserMock ).toHaveBeenCalledWith({
      name: 'Saulo Román', 
      email: 'romansantillan1998@outlook.com', 
      age: 26, 
    })

  })

})