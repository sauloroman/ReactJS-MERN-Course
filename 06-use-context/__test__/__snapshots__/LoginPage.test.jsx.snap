// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in LoginPage Component it must match with snapshot 1`] = `
<div>
  <div>
    <h1
      class="heading__page"
    >
      Bienvenido a la app
    </h1>
    <button
      class="btn"
    >
      Iniciar Sesi√≥n
    </button>
  </div>
</div>
`;
